store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedStore.java

```diff
@@ -362,31 +361,36 @@ public List<Graph> getGraphs(final User user, final List<String> graphIds, final
         return rtn;
     }
 
-    private String getKeyForProcessedFedStore() {
+    private String getKeyForProcessedFedStoreId() {
         return FEDERATED_STORE_PROCESSED + id;
     }
 
-    private boolean addFedStoreId(final Operation operation, final String optionKey) {
-        boolean hasOperationOrPayloadPreexistingFedStoreId = false;
-        if (nonNull(operation) && !isNullOrEmpty(optionKey)) {
-            //Keep Order v
-            boolean hasOperationPreexistingFedStoreId = !isNullOrEmpty(operation.getOption(optionKey, null)); //There is a difference between value null and key not found.
-            //Keep Order ^
-            boolean hasPayloadPreexistingFedStoreId = false;
-            if (operation instanceof FederatedOperation) {
-                //Check and Add FedStoreId to payload
-                hasPayloadPreexistingFedStoreId = addFedStoreId(((FederatedOperation) operation).getUnClonedPayload(), optionKey);
-            }
+    private boolean addFedStoreIdToOperation(final Operation operation) {
+        final String keyForFedStoreId = getKeyForProcessedFedStoreId();
+        boolean isFedStoreIdPreexisting = false;
+        if (nonNull(operation) && !isNullOrEmpty(keyForFedStoreId)) {
+            //KEEP THIS ORDER v
```

Comment:
I think perhaps it would be a bit better to explain this whole block needs to stay in the same order and maybe why, rather than all caps